dill/_dill.py

```diff
@@ -1059,6 +1059,54 @@ def _create_namedtuple(name, fieldnames, modulename, defaults=None):
         t = collections.namedtuple(name, fieldnames, defaults=defaults, module=modulename)
         return t
 
+# TODO: Remove this copy when #450 is pulled
+# https://github.com/python/cpython/blob/a8912a0f8d9eba6d502c37d522221f9933e976db/Lib/pickle.py#L322-L333
+def _getattribute(obj, name):
+    for subpath in name.split('.'):
+        if subpath == '<locals>':
+            raise AttributeError("Can't get local attribute {!r} on {!r}"
+                                 .format(name, obj))
+        try:
+            parent = obj
+            obj = getattr(obj, subpath)
+        except AttributeError:
+            raise AttributeError("Can't get attribute {!r} on {!r}"
+                                 .format(name, obj)) # In Py3: from None
+    return obj,
+
+def _create_capsule(pointer, name, context, destructor):
+    attr_found = False
+    try:
+        # based on https://github.com/python/cpython/blob/f4095e53ab708d95e019c909d5928502775ba68f/Objects/capsule.c#L209-L231
+        if PY3:
+            uname = name.decode('utf8')
+        else:
+            uname = name
+        for i in range(1, uname.count('.')+1):
+            names = uname.rsplit('.', i)
+            try:
+                module = __import__(names[0])
+            except:
+                pass
+            obj = module
+            for attr in names[1:]:
+                obj = getattr(obj, attr)
+            capsule = obj
+            attr_found = True
+            break
+    except:
+        pass
+
+    if attr_found:
+        if _PyCapsule_IsValid(capsule, name):
+            return capsule
+        raise UnpicklingError("%s object exists at %s but a PyCapsule object was expected." % (type(capsule), name))
+    else:
+        warnings.warn('Creating a new PyCapsule %s for a C data structure that may not be present in memory. Segmentation faults or other memory errors are possible.' % (name,), UnpicklingWarning)
```

Comment:
I'm on the fence about the use of a warning. You also added warnings in a previous PR related to pickling `numpy` objects. Warnings can be annoying, especially when you are running the code in parallel and they can be triggered on each new process.

Reply:
That's true. Maybe it would be nice to have `dill.settings['warnings']` which does exactly what you show above.  What do you think of that?

Reply:
I do not know whether it would be useful to do it globally or per pickler. Using the warnings package does make it a little bit more flexible allowing users to turn the warnings into errors, ignore all but the first time that the warning appears as well as filter some dill warnings but not others.

We could add the `'warnings'` setting and it would just change the filter to `"ignore"` if False or `"default"` if True as a shortcut while allowing others to use the functionality in the warnings package.

Reply:
let's not worry about it now.  But something to think about